routing/routing_session.cpp

```diff
@@ -582,9 +612,12 @@ shared_ptr<Route> const RoutingSession::GetRoute() const
 
 void RoutingSession::OnTrafficInfoClear()
 {
-  threads::MutexGuard guard(m_routingSessionMutex);
-  UNUSED_VALUE(guard);
-  Clear();
+  {
+    threads::MutexGuard guard(m_routingSessionMutex);
+    UNUSED_VALUE(guard);
+    Clear();
+  }
+  RebuildRouteOnTrafficUpdate();
```

Comment:
As far as I know this method may be called from any thread. Do we have a data race on callback fields?